orchagent/neighorch.cpp

```diff
@@ -226,7 +291,7 @@ bool NeighOrch::removeNeighbor(NeighborEntry neighborEntry)
     }
 
     m_syncdNeighbors.erase(neighborEntry);
-    m_routeOrch->removeNextHopEntry(ip_address);
+    removeNextHop(ip_address);
```

Comment:
line 279-290, why do we need to create the neighbor entry if remove is not successful?